src/components/icon/types.ts

```diff
@@ -0,0 +1,7 @@
+import type { SVGAttributes } from "react";
+
+export type IconSize = "small" | "large" | "medium";
+
+export interface DefaultIconProps extends SVGAttributes<SVGElement> {
+  size?: number;
```

Comment:
i thought size can be a number or IconSize. why is this here only a number?

why is this props interface outside of the component?

Reply:
Its used by two different components,

Icon (is the wrapper that resolves to the individual icon,  converts the size to a number)
The individual icons accept size as number only, no need the run the getter here when Icon already exists